contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,850 @@
+
+### Out of Scope
+- As the administrator of a stateful application, in order to increase the 
+applications storage capacity, I want to update PersistentVolumes.
+- As the administrator of a stateful application, in order to update the 
+network configuration of the application, I want to update Services and 
+container ports in a consistent way.
+- As the administrator of a stateful application, when I scale my application 
+horizontally, I went associated PodDistruptionBudgets to be adjusted to 
+compensate for the application's scaling.
+
+## Assumptions
+ - StatefulSet update must support singleton StatefulSets. However, an update in
+ this case will cause a temporary outage. This is acceptable as a single 
+ process application is, by definition, not highly available.
+ - Disruption in Kubernetes is controlled by PodDistruptionBugets. As 
+ StatefulSet updates progress one Pod at a time, and only occur when all 
+ other Pods have a Status of Running and a Ready Condition, they can not 
+ violate reasonable PodDisrutptionBugdets.
+ - Without priority and preemption, there is no guarantee that an update will 
+ not block due to a loss of capacity or due to the scheduling of another Pod
+ between Pod termination and Pod creation. This is mitigated by blocking the 
+ update when a Pod fails to schedule. Remediation will require operator 
+ intervention. This implementation is no worse than the current behavior with 
+ respect to eviction.
+ - We will eventually implement a signal that is delivered to Pods to indicate 
+ the 
+ [reason for termination](https://github.com/kubernetes/kubernetes/issues/1462).
+ This will be a general implementation, usable for any Pod in a Kubernetes 
+ cluster. It is, therefore, out of scope to design such a mechanism here.
+ - Kubelet does not support resizing a container's resources without terminating 
+ the Pod. In place resource reallocation is out of scope for this design. 
+ Vertical scaling must be performed destructively.
+ - The primary means of configuration update will be configuration files, 
+ command line flags, environment variables, or ConfigMaps consumed as the one 
+ of the former. 
+ - In place configuration update via SIGHUP is not universally 
+ supported, and Kubelet provides no mechanism to perform this currently. Pod 
+ reconfiguration will be performed destructively.
+ - Stateful applications are likely to evolve wire protocols and storage formats
+  between versions. In most cases, when updating the application's Pod's 
+  containers, it will not be safe to roll back or forward to an arbitrary 
+  version. StatefulSet update should work well when rolling out an update, 
+  or performing a rollback, between two specific revisions of the StatefulSet.
+
+## Requirements
+This design is based on the following requirements.
+- Users must be able to update the containers of a StatefulSet's Pods.
+  - Updates to container commands, images, resources and configuration must be 
```

Comment:
The changes we are considering now would be mutation to the VolumeClaimsTemplate with respect to resizing the storage. It would not be an identity change because we would not mutate the contents of the storage.